### PR TITLE
Add Set-AzureRmMlOpCluster cmdlet

### PR DESCRIPTION
**Pull request by shutchings**
## Description
Added the Set-AzureRmMlOpCluster cmdlet. Also updated the .NET SDK to 0.2.0.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

File: setup/azurecmdfiles.wxi [hunk 1]
```diff
@@ -759,15 +759,75 @@
                         <Component Id="cmpB2597D237191FA5EFBD36804D217035B" Guid="*">
                             <File Id="fil943575A5C76A6FE22FF176958FCBD7FD" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.MachineLearningCompute\AzureRM.MachineLearningCompute.psm1" />
                         </Component>
+                        <Component Id="cmp8DCB9196E6A234B735C6511571A39F87" Guid="*">
```
Comment (cormacpayne) [inline setup/azurecmdfiles.wxi]: @shutchings it looks like these extra dlls didn't get deleted during the full build. There is a `CleanupBuild.ps1` script that runs at the end of `msbuild build.proj` that removes these redundant assemblies in your output folder. Would you mind regenerating this file after a full build?

File: src/ResourceManager/MachineLearningCompute/AzureRM.MachineLearningCompute.psd1 [hunk 2]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '0.2.0'
+ModuleVersion = '0.3.0'
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearningCompute/AzureRM.MachineLearningCompute.psd1]: @shutchings you can revert the changes in this PR that bump the module version - we will handle that with separate tooling

File: src/ResourceManager/MachineLearningCompute/ChangeLog.md [hunk 3]
```diff
@@ -19,6 +19,13 @@
 -->
 ## Current Release
 
+## Version 0.3.0
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearningCompute/ChangeLog.md]: @shutchings please remove this `## Version 0.3.0` header and move the below contents to be under `## Current Release`

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+using Microsoft.Azure.Commands.MachineLearningCompute.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.MachineLearningCompute;
+using Microsoft.Azure.Management.MachineLearningCompute.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.MachineLearningCompute.Cmdlets
+{
+    [Cmdlet(VerbsCommon.Set, CmdletSuffix, SupportsShouldProcess = true)]
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs]: @shutchings make sure that this cmdlet has a `DefaultParameterSetName`, which will be the `SetByIndividualParameters` set so users are prompted to provide `Name` and `ResourceGroupName`

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+            HelpMessage = SslCNameParameterHelpMessage)]
+        [Parameter(ParameterSetName = ResourceIdParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = SslCNameParameterHelpMessage)]
+        public string SslCName { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Name, @"Updating operationalization cluster..."))
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs]: @shutchings I would move the two if-statements that set `Name` and `ResourceGroupName` for the `ObjectParameterSet` and `ResourceIdParameterSet` sets above this line so that `Name` has an appropriate value before being printed by the `ShouldProcess` block

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Commands.MachineLearningCompute.csproj [hunk 6]
```diff
@@ -65,14 +65,18 @@
       <Name>Commands.ResourceManager.Common</Name>
     </ProjectReference>
     <Reference Include="Microsoft.Azure.Management.MachineLearningCompute, Version=0.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.MachineLearningCompute.0.1.0\lib\net452\Microsoft.Azure.Management.MachineLearningCompute.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.MachineLearningCompute.0.2.0\lib\net452\Microsoft.Azure.Management.MachineLearningCompute.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL" />
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL" />
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL" />
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Commands.MachineLearningCompute.csproj]: @shutchings please remove the above references that were added (`Microsoft.Rest.*` and `Newtonsoft.Json`). These are already defined in the `Common.Dependencies.targets` file.

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/packages.config [hunk 7]
```diff
@@ -1,4 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.Management.MachineLearningCompute" version="0.1.0" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.MachineLearningCompute" version="0.2.0" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.10" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.10" targetFramework="net452" />
+  <package id="Newtonsoft.Json" version="6.0.8" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/packages.config]: @shutchings same comment about removing the added references

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute.Test/ScenarioTests/MLCTests.ps1 [hunk 8]
```diff
@@ -70,6 +66,15 @@ function TeardownTest([String] $ResourceGroupName)
     Write-Debug "Delete resource group"
     Write-Debug " Resource Group Name : $resourceGroupName"
     Remove-AzureRmResourceGroup -Name $ResourceGroupName -Force
+
+	foreach ($g in Find-AzureRmResourceGroup)
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute.Test/ScenarioTests/MLCTests.ps1]: @shutchings `Get-AzureRmResourceGroup`

File: src/ResourceManager/MachineLearningCompute/AzureRM.MachineLearningCompute.psd1 (hunk 2 above)
Comment (shutchings) [reply src/ResourceManager/MachineLearningCompute/AzureRM.MachineLearningCompute.psd1]: Done.

File: src/ResourceManager/MachineLearningCompute/ChangeLog.md (hunk 3 above)
Comment (shutchings) [reply src/ResourceManager/MachineLearningCompute/ChangeLog.md]: Done.

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs (hunk 4 above)
Comment (shutchings) [reply src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs]: Sounds good.

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs (hunk 5 above)
Comment (shutchings) [reply src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs]: Good catch.

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Commands.MachineLearningCompute.csproj (hunk 6 above)
Comment (shutchings) [reply src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Commands.MachineLearningCompute.csproj]: Got it.

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/packages.config (hunk 7 above)
Comment (shutchings) [reply src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/packages.config]: Got it.

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute.Test/ScenarioTests/MLCTests.ps1 (hunk 8 above)
Comment (shutchings) [reply src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute.Test/ScenarioTests/MLCTests.ps1]: Couldn't figure out how to get this to work. Instead I'm getting the resource group a different way.

**Comment by shutchings** [comment]
@cormacpayne The unit tests are working now. I'm still having trouble with the build/installer though. It keeps having problems with the Automapper dependency but I'm not sure why.

File: src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+        protected const string CmdletParametersParameterSet = "SetByIndividualParameters";
+
+        [Parameter(ParameterSetName = ObjectParameterSet,
+            Mandatory = true, 
+            ValueFromPipeline = true,
+            HelpMessage = ClusterParameterHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        [Alias(ClusterInputObjectAlias)]
+        public PSOperationalizationCluster InputObject { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs]: @shutchings had a small internal discussion about what to do with `InputObject` in the piping scenario for `Set-*` cmdlets, and we decided that the `InputObject` parameter set should have all of the property parameters as the other sets, and if the user passes a value for any property parameter, then you update the corresponding property of the `InputObject` before passing it to the server.

Comment (shutchings) [reply src/ResourceManager/MachineLearningCompute/Commands.MachineLearningCompute/Cmdlets/SetAzureRmMlOpCluster.cs]: That makes sense. Check out the latest changes.

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/241/

**Comment by shutchings** [comment]
@cormacpayne can I build a package job to test the install?

**Comment by cormacpayne** [comment]
Packaging job: https://azuresdkci.westus2.cloudapp.azure.com/job/posh-pack/40/
